src/main/java/com/es/phoneshop/model/product/cart/DefaultCartService.java

```diff
@@ -38,6 +40,18 @@ public synchronized void add(Cart cart, Long productId, int quantity) throws Out
         if (product.getStock() < quantity) {
             throw new OutOfStockException(product, quantity, product.getStock());
         }
-        cart.getItems().add(new CartItem(product, quantity));
+        CartItem cartItem = new CartItem(product, quantity);
+
+        if (cart.getItems().contains(cartItem)) {
```

Comment:
`cart.getItems().contains(cartItem)`
`cart.getItems().stream().filter(....)`
you basically check if product exist 2 times instead of 1